imap_processing/ultra/l1b/extendedspin.py

```diff
@@ -0,0 +1,39 @@
+"""Calculates Extended Spin."""
+
+import numpy as np
+
+
+def calculate_extendedspin(data_dict):
+    """
+    Create dictionary with defined datatypes for Extended Spin Data.
+
+    Parameters
+    ----------
+    data_dict: : dict
+        L1a data dictionary.
+
+    Returns
+    -------
+    extendedspin_dict : dict
+        Dictionary containing the data.
```

Comment:
Can you make this a dataset instead of a dictionary? Each of your keys below would be separate data arrays.

Could also loop over the float and uint values.
```python
for name in ["spin_number", "spin_start_time", ...]:
    da = xr.DataArray((name, epoch, np.zeros(len(epoch), dtype=np.uint64)))
```